### PR TITLE
Bump cairo lang to 0.11.1.1 and starknet-devnet to 0.5.2

### LINKED ISSUE #1872: Sync with cairo-lang 0.11.1
We need to prepare for 0.11.1 update supporting 1.0 compiler version, pre release is already available
https://github.com/starkware-libs/cairo-lang/compare/v0.11.0.2...master

### PR DESCRIPTION
**Pull request by abulenok**
### Checklist
- [x] Relevant issue is linked
- [x] Docs updated/issue for docs created
- [ ] Added relevant tests

### CODE REVIEW

**Comment by abulenok** [comment]
At this moment, tests that are using devnet fail. To make them pass we need a newer version of devnet with cairo-lang updated to 0.11.1

**Review by karol-bisztyga — COMMENTED** [review]
Ok, I looked at these changes and didn't spot any obvious mistakes. That said, I assume all this is just some refactoring so everything works with a new version, is that correct?

Also a question: why there's no version bump in `pyproject.toml`? I know this is "prepare for" but I'm just curious. Also, we probably don't want to merge this until we bump the version, right?

**Comment by abulenok** [comment]
@karol-bisztyga Yes, this is mainly a refactoring, the only change of the logic was introduced in `protostar/testing/starkware/contract_based_test_execution_state.py`.

As of `pyproject.toml`, I think when I started to work on this task, `0.11.1a0` was not available on PyPI, so I had to install it with pip. But yeah we can put it into `pyproject.toml` now

> Also, we probably don't want to merge this until we bump the version, right?

Yes, I think we should wait for the official release

**Review by karol-bisztyga — CHANGES_REQUESTED** [review]
> Yes, I think we should wait for the official release

Ok, I'm going to request changes just to make sure

**Comment by abulenok** [comment]
The `--profiling` argument is commented out due to an error in import statement https://github.com/software-mansion/protostar/blob/14558019c151121be4ab7f6dd67b12347b3a144c/protostar/profiler/pprof.py#L4

```
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
```

`starknet-devnet` requires `web3` 6.0.0, which brings a newer version of `protobuf`. That's why the error occurs

File: protostar/commands/legacy_commands/test_cairo0/test_cairo0_command.py
Comment (MaksymilianDemitraszek) [inline protostar/commands/legacy_commands/test_cairo0/test_cairo0_command.py:109]: Let's just remove it, we won't be updating it

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM, Let's just remove everything profiling related before merging

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM

**Review by karol-bisztyga — APPROVED** [review]

**Pull request merged**
